### ISSUE TITLE
Crashes when opening a plugin that wants to render using gtk when suil isn't compiled with gtk

### ISSUE THREAD

**Issue by simonvanderveldt** [rncbc/qtractor#47]
Qtractor hard crashes when I add a plugin that (apparently) wants to render it's UI using GTK. When enabling debugging I get the following error message:

```
suil error: Unable to open wrap module /usr/lib64/suil-0/libsuil_gtk2_in_qt4.so (/usr/lib64/suil-0/libsuil_gtk2_in_qt4.so: cannot open shared object file: No such file or directory)
```

This is easily solved by compiling suil with gtk support, but I do think that Qtractor shouldn't crash when it can't find a lib. Is there a way to prevent this?

Also, there seem to be some checks to prevent these situations in [`configure.ac`](https://github.com/rncbc/qtractor/blob/master/configure.ac#L1251) but these seem to explicitly only check when not using Qt4. That might be an error?
(For reference: I'm running a Qt4 build)

**Comment by rncbc** [comment]
On 09/25/2016 04:55 PM, Simon van der Veldt wrote:
> Qtractor hard crashes when I add a plugin that (apparently) wants to
> render it's UI using GTK. When enabling debugging I get the following
> error message:
>
> |suil error: Unable to open wrap module
> /usr/lib64/suil-0/libsuil_gtk2_in_qt4.so
> (/usr/lib64/suil-0/libsuil_gtk2_in_qt4.so: cannot open shared object
> file: No such file or directory) |
>
> This is easily solved by compiling suil with gtk support, but I do think
> that Qtractor shouldn't crash when it can't find a lib. Is there a way
> to prevent this?
>

libsuil_gtk2_in_qt4 object or specific package must be installed on your 
system; it is part of suil, not of qtractor.


> Also, there seem to be some checks to prevent these situations in
> |configure.ac|
> <https://github.com/rncbc/qtractor/blob/master/configure.ac#L1251> but
> these seem to explicitly only check when not using Qt4. That might be an
> error?
> (For reference: I'm running a Qt4 build)
>

yes, those checks are for an alternative to libsuil_gtk2_in_qt5, which 
is only available on suil >= 0.8.4 but still no official announcement of 
this latest suil release.

byee
-- 
rncbc aka. Rui Nuno Capela

**Comment by simonvanderveldt** [comment]
> libsuil_gtk2_in_qt4 object or specific package must be installed on your
> system; it is part of suil, not of qtractor.

Yeah, that's what I did, maybe I worded it in an unclear way. I recompiled libsuil with gtk enabled.
But I do feel that it shouldn't cause a segfault in Qtractor. I'm able to figure out what's happening by creating a debug build, but for people who know less about these things it's not really a good user experience.
Or is it impossible to catch these types of errors?

> yes, those checks are for an alternative to libsuil_gtk2_in_qt5, which
> is only available on suil >= 0.8.4 but still no official announcement of
> this latest suil release.

Ah, I see. And that alternative is Qt5 specific?

**Comment by rncbc** [comment]
On 09/25/2016 08:13 PM, Simon van der Veldt wrote:
>     libsuil_gtk2_in_qt4 object or specific package must be installed on your
>     system; it is part of suil, not of qtractor.
>
> Yeah, that's what I did, maybe I worded it in an unclear way. I
> recompiled libsuil with gtk enabled.
> But I do feel that it shouldn't cause a segfault in Qtractor. I'm able
> to figure out what's happening by creating a debug build, but for people
> who know less about these things it's not really a good user experience.
> Or is it impossible to catch these types of errors?
>
afaict. it seems a packaging/install/deployment error: you have to 
install ALL the objects or packages that were ganged on same libsuil 
build. otherwise you get segfault/crashes due to missing installed 
components (the API assumes is there, but on the file-system isn't).


>     yes, those checks are for an alternative to libsuil_gtk2_in_qt5, which
>     is only available on suil >= 0.8.4 but still no official announcement of
>     this latest suil release.
>
> Ah, I see. And that alternative is Qt5 specific?
>

yes. exactly.

byee
-- 
rncbc aka. Rui Nuno Capela

**Comment by simonvanderveldt** [comment]
@rncbc I'm not sure I fully understand. Do you mean the build system for Qtractor does build-time checks if libraries exist/have certain feature enabled?
Because the libsuil I had and used to build Qtractor didn't have gtk enabled but the compiled Qtractor apparently still expected it to be there.

**Comment by rncbc** [comment]
> Do you mean the build system for Qtractor does build-time checks if libraries exist/have certain feature enabled?

not that if libraries exist but it checks on the API with suil_ui_supported() call that tells whether some wrapper or library was compiled in at libsuil build time, and is featured or supported (eg. libsuil_qtk2_in_qt4.so . however, it seems, that although it might tell it's supported (was built at a former time), it might not be actually deployed (installed) on the system and the missing is only evident at runtime, when the host (qtractor) goes to actually show up a plugin's GtkUI and is greeted with a crash (eg. libsuil_gtk2_in_qt4.so not found).

byee
-- 
rncbc aka. Rui Nuno Capela

**Comment by simonvanderveldt** [comment]
> yes, those checks are for an alternative to libsuil_gtk2_in_qt5, which
> is only available on suil >= 0.8.4 but still no official announcement of
> this latest suil release.

There has been a release of suil 0.10.0 now :)
Is there anything needed to use this or will it just work by updating suil and rebuilding Qtractor?

**Comment by rncbc** [comment]
unless you're using kxstudio repos--which tend to use libsuil-static and thus qtractor must also get updated--you're all fine to deploy libsuil 0.10.0 and go.

byee
